packages/yourloops/components/consent.tsx

```diff
@@ -0,0 +1,272 @@
+      alignItems: "center",
+    },
+    Card: {
+      display: "flex",
+      flexDirection: "column",
+      justifyContent: "center",
+      textAlign: "center",
+      // eslint-disable-next-line no-magic-numbers
+      padding: theme.spacing(4),
+    },
+    CardContent: {
+      marginLeft: theme.spacing(2),
+      marginRight: theme.spacing(2),
+    },
+    FormControl: {
+      margin: theme.spacing(3),
+    },
+    FormHelperText: {
+      textAlign: "center",
+    },
+    FormControlLabel: {
+      alignItems: "start",
+      textAlign: "start",
+      marginTop: theme.spacing(2),
+      marginBottom: theme.spacing(2),
+      marginLeft: theme.spacing(1),
+      marginRight: theme.spacing(1),
+    },
+    Buttons: {
+      display: "flex",
+      flexDirection: "row",
+      marginTop: theme.spacing(2),
+      marginRight: theme.spacing(5), // eslint-disable-line no-magic-numbers
+      marginLeft: theme.spacing(1), // eslint-disable-line no-magic-numbers
+    },
+    Button: {
+      marginLeft: "auto",
+    },
+  };
+});
+
+/**
+ * Patient Consent Page
+ */
+function Consent(props: ConsentProps): JSX.Element {
+  const { t, i18n } = useTranslation("yourloops");
+  const auth = useAuth();
+  const historyHook = useHistory();
+  const classes = style();
+  const [terms, setTerms] = React.useState(false);
+  const [privacyPolicy, setPrivacyPolicy] = React.useState(false);
+  const [error, setError] = React.useState(false);
+  const [helperText, setHelperText] = React.useState("");
+
+  const linkTermsText = t("Terms of Use");
+  const linkTerms = DiabeloopUrl.getTermsLink(i18n.language);
+  const privacyPolicyText = t("Privacy Policy");
+  const linkPrivacyPolicy = DiabeloopUrl.getPrivacyPolicyLink(i18n.language);
+
+  const resetFormState = (): void => {
+    setError(false);
+    setHelperText("");
+  };
+
+  const onChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>,
+    setState: React.Dispatch<React.SetStateAction<boolean>>
+  ): void => {
+    setState((event.target as HTMLInputElement).checked);
+    resetFormState();
+  };
+
+  const valideForm = (): boolean => {
+    if (terms && privacyPolicy) {
+      return true;
+    }
+
+    setError(true);
+    setHelperText("required");
+    return false;
+  };
+
+  const onDecline = (/* event: React.MouseEvent<HTMLButtonElement, MouseEvent> */) => {
+    auth.logout();
+  };
+
+  const onConfirm = (/* event: React.MouseEvent<HTMLButtonElement, MouseEvent> */) => {
+    resetFormState();
+    if (valideForm()) {
+      // api call
+      const user = auth?.user;
+      if (user?.profile !== undefined && user?.profile !== null) {
+        const now = new Date().toISOString();
+        user.profile.termsOfUse = { acceptanceTimestamp: now, isAccepted: terms };
+        user.profile.privacyPolicy = { acceptanceTimestamp: now, isAccepted: privacyPolicy };
+        auth.updateProfile(user.profile);
+        historyHook.push(props.destinationPath);
+      }
```

Comment:
We should update the profile even if it not exists yet.
I know, this should not happen.

```suggestion
      const user = auth.user;
      if (user === null) {
        throw new Error("User must be logged-in");
      }
      const now = new Date().toISOString();
      const updatedProfile = _.cloneDeep(user.profile ?? {}) as Profile;
      updatedProfile.termsOfUse = { acceptanceTimestamp: now, isAccepted: terms };
      updatedProfile.privacyPolicy = { acceptanceTimestamp: now, isAccepted: privacyPolicy };
      auth.updateProfile(updatedProfile).catch((reason: unknown) => {
        console.error(reason);
      }).finally(() => {
        historyHook.push(props.destinationPath);
      });
```

Of course some imports are missing with my suggestion.